src/app/GitCommands/Git/AheadBehindDataProvider.cs

```diff
@@ -96,7 +96,7 @@ public void ResetCache()
             foreach (Match match in matches)
             {
                 string branch = match.Groups["branch"].Value;
-                string remoteRef = (match.Groups["remote_p"].Success && !string.IsNullOrEmpty(match.Groups["remote_p"].Value))
+                string remoteRef = (match.Groups["remote_p"].Success && !string.IsNullOrEmpty(match.Groups["remote_p"].Value) && !match.Groups["gone_p"].Success)
                     ? match.Groups["remote_p"].Value
                     : match.Groups["remote_u"].Value;
```

Comment:
A comment would be helpful here and / or at `AheadBehindRegex`.
(You are just using "remote_p" / "remote_u", which should have been explained when first introduced.)
Without deeper analysis, I cannot assess whether this condition might be too restrictive.

Reply:
➡️ Added a comment